code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,465 @@
+	var/faction = list("hostile")
+	///Spawner component
+	var/spawner_type = /datum/component/spawner
+	///This var check if the portal has been closed by a player with a neutralizer
+	var/closed = FALSE
+
+/obj/structure/crystal_portal/Initialize()
+	. = ..()
+	AddComponent(spawner_type, mob_types, spawn_time, faction, spawn_text, max_mobs)
+	GLOB.crystal_portals += src
+
+/obj/structure/crystal_portal/Destroy()
+	GLOB.crystal_portals -= src
+	if(!closed)
+		switch(name)
+			if("Small Portal")
+				explosion(loc, 0,1,3)
+			if("Medium Portal")
+				explosion(loc, 0,3,5)
+			if("Big Portal")
+				explosion(loc, 1,3,5)
+			if("Huge Portal")
+				explosion(loc, 2,5,7)
+	new/obj/item/assembly/signaler/anomaly(loc)
+	return ..()
+
+/obj/structure/crystal_portal/attack_animal(mob/living/simple_animal/M)
+	if(faction_check(faction, M.faction, FALSE)&&!M.client)
+		return
+	..()
+
+/obj/structure/crystal_portal/attackby(obj/item/W, mob/living/user, params)
+	if(!istype(W) || (W.item_flags & ABSTRACT) || !istype(user))
+		return
+	if(istype(W, /obj/item/anomaly_neutralizer))
+		to_chat(user, "<span class='notice'>You start closing the [src]</span>")
+		if(W.use_tool(src, user, 5.5 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>You you successfully close the [src]</span>")
+			closed = TRUE
+			qdel(src)
+
+/obj/structure/crystal_portal/small
+	name = "Small Portal"
+	desc = "A small portal to an unkown dimension!"
+	color = "#B2FFFE"
+	max_mobs = 3
+	spawn_time = 200
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug)
+
+/obj/structure/crystal_portal/medium
+	name = "Medium Portal"
+	desc = "A medium portal to an unkown dimension!"
+	color = "#93F3B2"
+	max_mobs = 5
+	spawn_time = 180
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit)
+
+/obj/structure/crystal_portal/big
+	name = "Big Portal"
+	desc = "A big portal to an unkown dimension!"
+	color = "#F4F48A"
+	max_mobs = 8
+	spawn_time = 160
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit, /mob/living/simple_animal/hostile/crystal_monster/killer)
+
+/obj/structure/crystal_portal/huge
+	name = "Huge Portal"
+	desc = "A huge portal to an unkown dimension!"
+	color = "#F97575"
+	max_mobs = 12
+	spawn_time = 140
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit, /mob/living/simple_animal/hostile/crystal_monster/killer, \
+					 /mob/living/simple_animal/hostile/crystal_monster/boss)
```

Comment:
```suggestion
	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion,
		/mob/living/simple_animal/hostile/crystal_monster/thug,
		/mob/living/simple_animal/hostile/crystal_monster/recruit,
		/mob/living/simple_animal/hostile/crystal_monster/killer,
		/mob/living/simple_animal/hostile/crystal_monster/boss,
		)
```
You don't need `\` unless if you are formatting text really.
Adding a comma on the last list member won't add a null entry and will diminish lines changed if a new one is introduced later.
This pattern is a bit more clear to me, but it's optional.